store.go

```diff
@@ -3103,13 +3109,21 @@ func (s *store) CleanupStagedLayer(diffOutput *drivers.DriverWithDifferOutput) e
 	return err
 }
 
+func (s *store) PrepareStagedLayer(options *drivers.ApplyDiffWithDifferOpts, differ drivers.Differ) (*drivers.DriverWithDifferOutput, error) {
+	return s.ApplyDiffWithDiffer("", options, differ)
+}
+
 func (s *store) ApplyDiffWithDiffer(to string, options *drivers.ApplyDiffWithDifferOpts, differ drivers.Differ) (*drivers.DriverWithDifferOutput, error) {
-	return writeToLayerStore(s, func(rlstore rwLayerStore) (*drivers.DriverWithDifferOutput, error) {
-		if to != "" && !rlstore.Exists(to) {
-			return nil, ErrLayerUnknown
-		}
-		return rlstore.ApplyDiffWithDiffer(to, options, differ)
-	})
+	rlstore, err := s.getLayerStore()
+	if err != nil {
+		return nil, err
+	}
+
+	if to != "" {
+		return nil, fmt.Errorf("ApplyDiffWithDiffer does not support non-empty 'layer' parameter")
+	}
+
+	return rlstore.applyDiffWithDifferNoLock(nil, options, differ)
```

Comment:
The first parameter can be entirely removed, as well as the relevant code paths down the call stack.

Reply:
thanks, fixed.

I'll work on a follow up to simplify the `ApplyDiffWithDiffer` in the graph driver too

Reply:
> I'll work on a follow up to simplify the `ApplyDiffWithDiffer` in the graph driver too

added a patch to this PR